frappe/modules/import_file.py

```diff
@@ -127,4 +127,20 @@ def import_doc(docdict, force=False, data_import=False, pre_process=None,
 		doc.flags.ignore_permissions = True
 		doc.flags.ignore_mandatory = True
 	doc.insert()
+	update_icon(doc)
 	frappe.flags.in_import = False
+
+def update_icon(doc):
```

Comment:
you should write a specific patch if you change an icon and update via `patches.txt`

this is a very low level method and not the right place for this.